src/win32/path_w32.c

```diff
@@ -110,11 +110,10 @@ int git_win32_path_canonicalize(git_win32_path path)
 
 		else if (len == 2 && from[0] == L'.' && from[1] == L'.') {
 			if (to == base) {
-				/* no more path segments to strip, eat the "../" */
+				/* Skip over leading "../" */
```

Comment:
This isn't a _leading_ `../`, though, at least not always.

`base` is the results of `path_skip_prefix`, which will skip over things that we never touch - like drive prefixes and UNC share prefixes.

So if we tried to leave these in place, then we'd leave them when they exist immediately behind the drive prefixes.

For example, `C:\..\foo\bar` _should_ be canonicalized as `C:\foo\bar` (just like you would canonicalize `/../foo/bar` as `/foo/bar`).  That's the goal of this bit.

Reply:
I think that we want to #5435